cmd/wazero/wazero.go

```diff
@@ -222,11 +223,28 @@ func doRun(args []string, stdOut io.Writer, stdErr io.Writer, exit func(code int
 		exit(1)
 	}
 
-	// WASI is needed to access args and very commonly required by self-contained wasm
-	// binaries, so we instantiate it by default.
-	wasi_snapshot_preview1.MustInstantiate(ctx, rt)
+	var needsWASI, needsGo bool
+IMPORTS:
+	for _, f := range code.ImportedFunctions() {
+		moduleName, _, _ := f.Import()
+		switch moduleName {
+		case wasi_snapshot_preview1.ModuleName:
+			needsWASI = true
+			break IMPORTS // can't be both WASI and go
+		case "go":
+			needsGo = true
+			break IMPORTS // can't be both WASI and go
+		}
+	}
```

Comment:
I think maybe extract this as a function, and make it accept []api.Function and return needsWASI and needsGo, then testable?

Reply:
done!